packages/calcite-components/src/utils/form.tsx

```diff
@@ -290,7 +290,7 @@ function invalidHandler(event: Event) {
         formComponent.status = "idle";
       }
 
-      if ("validationIcon" in formComponent && !formComponent.validationIcon) {
+      if ("validationIcon" in formComponent) {
```

Comment:
Unless I am wrong, it seems like this condition should not have been here and it worked in Stencil accidentally:

- Since validationIcon prop is of mixed type (string | boolean), setting it to empty string does not automatically cast to `true`
- You set validationIcon prop to true here on form validation error:
  - https://github.com/Esri/calcite-design-system/blob/fed772f8518b9d3b2190f62263b6de383ea45dc3/packages/calcite-components/src/utils/form.tsx#L281
- SInce this prop is reflected, stencil reflects it to empty string attribute:
  - https://github.com/ionic-team/stencil/blob/61f90b04ecf8896d74840ea295f949f3de38676c/src/runtime/vdom/set-accessor.ts#L178
- Reflection of the attribute triggers attributeChangedCallback. Stencil sees that the attribute value `""` does not match the property value `true`, and sets the property to `""`.
- When it comes to clear the validation message, this place does the `&& !formComponent.validationIcon` check. That worked because `""` is falsy.
  - However, this condition does not work in Lit because Lit ignores attributeChangedCallback when it does the reflection of a property to an attribute

Reply:
IIRC, this logic preserves user-set validation icons between invalid form submissions. You can see the behavior in [this pen](https://codepen.io/jcfranco/pen/abepQWE?editors=1000).

@benelan could you confirm? If so, could you look at updating `formAssociated` to cover this case?

Reply:
> IIRC, this logic preserves user-set validation icons between invalid form submissions. 

Yeah this is correct. I definitely abused some insane javascript quirks here so that's my bad lol. If you need help refactoring this so it will work in Lit I can take another look.

> If so, could you look at updating formAssociated to cover this case?

Yeah definitely.

Reply:
Thanks for the fix, @benelan!